src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterActivator.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.AspNet.Mvc.Core;
+using Microsoft.Framework.DependencyInjection;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class DefaultFilterActivator : IFilterActivator
+    {
+        private readonly IServiceProvider _provider;
+        private static readonly ConcurrentDictionary<Type, ObjectFactory> _filterCache =
+               new ConcurrentDictionary<Type, ObjectFactory>();
+
+        public DefaultFilterActivator([NotNull] IServiceProvider provider)
```

Comment:
`provider` -> either to `serviceProvider` or `services`. provider by itself is meaningless